cpp/oneapi/dal/algo/subgraph_isomorphism/backend/cpu/matching.hpp

```diff
@@ -350,65 +356,70 @@ solution<Cpu> matching_engine<Cpu>::get_solution() {
     return std::move(engine_solutions);
 }
 
+template <typename Cpu>
+std::int64_t matching_engine<Cpu>::get_match_count() const {
+    return engine_solutions.get_solution_count();
+}
+
+template <typename Cpu>
+std::int64_t matching_engine<Cpu>::state_exploration() {
+    if (target->bit_representation) {
+        return state_exploration_bit();
+    }
+    else {
+        return state_exploration_list();
+    }
+}
+
 template <typename Cpu>
 void matching_engine<Cpu>::run_and_wait(global_stack<Cpu>& gstack,
                                         std::int64_t& busy_engine_count,
+                                        std::int64_t& current_match_count,
+                                        std::int64_t target_match_count,
                                         bool main_engine) {
     if (main_engine) {
         first_states_generator(hlocal_stack);
     }
     bool is_busy_engine = true;
     ONEDAL_ASSERT(pattern != nullptr);
-    if (target->bit_representation) { /* dense graph case */
```

Comment:
Have you checked that 'if' in every state_exploration() won't affect performance?

Reply:
I have checked, it has no effect to performance